### PR TITLE
Fix environment visualizer frame issues 

### PR DESCRIPTION
**Pull request by CrownKira**
### Description

**Fix frame arrows overlap**
_Before:_
Screenshot 2021-01-17 at 2 14 56 PM
_After:_
Screenshot 2021-01-17 at 2 12 10 PM

Others:
- Fix frame titles overlap
- Fix issues visualizing the environments of chained functions (currently doesn't support visualization of unbound functions)

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Code quality improvements

### How to test

```javascript
let x = null;
for (let i = 0; i < 5; i = i + 1) {
    x = pair(() => i, x);
}

debugger;
```
```javascript
(w => x => y => { const a = 1; debugger; })(1)(2)(3);
```

### Checklist

- [x] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

**Review by angelsl — APPROVED** [review]
Lightly tested, seems to work. I don't have time to do a thorough code review, but I will leave r+ anyway.

**Comment by angelsl** [comment]
Actually, it might be better if you get someone else (who has reviewed the code) to give you r+.

File: public/externalLibs/env_visualizer/visualizer.js [hunk 1]
```diff
@@ -9,26 +9,23 @@
...
   container.style.width = 0;
   document.body.appendChild(container);
-  // create viewport
-  const viewport = new Concrete.Viewport({
-    container: container
-  });
+  const viewport = new Concrete.Viewport({ container });
+
+  const { PRODUCTION_MODE = true } = exports;
+  const DEBUG_MODE = !PRODUCTION_MODE && true; // enable to see debug messages in development
```
[the remark this answers is not in the document]
Comment (CrownKira) [reply public/externalLibs/env_visualizer/visualizer.js:15]: We can set it to '&& false' to turn off debug_mode in development.

File: public/externalLibs/env_visualizer/visualizer.js [hunk 2]
```diff
@@ -1805,55 +1710,130 @@
...
+        initialiseTextObject(frameName, x, y - 10, { color: WHITE, maxWidth: 100, isSymbol: true })
+      );
+
+      // render text in frame
+      const textConfig = { color: WHITE, maxWidth: Infinity, isSymbol: true };
+      let i = 0;
+      let textX = x + FRAME_PADDING_LEFT;
+      let textY = y + FRAME_PADDING_TOP;
+
+      for (const name in elements) {
```
[the remark this answers is not in the document]
Comment (CrownKira) [reply public/externalLibs/env_visualizer/visualizer.js]: I have made the change in this commit: 0584b21716f8154def175dfa4c9c788a1f571a99

**Review by jeffsieu — APPROVED** [review]
lgtm!

**Pull request merged**